### PR TITLE
Validate labels in `compute_loss` call for `TensorFlowV2Classifier` and `KerasClassifier`

### LINKED ISSUE #1959: TensorFlowV2Classifier and KerasClassifier do not validate labels when computing loss
**Describe the bug**
In the current implementations for the `TensorFlowV2Classifier` and `KerasClassifier`, when calling the `compute_loss(x, y)` method, the labels are not validated and transformed. This causes an error when the `argmax` function is called and integer class labels are passed.

The `PyTorchClassifier` class performs this validation so the error is avoided. The fix is relatively straightforward, just call the `check_and_transform_label_format` function in the `TensorFlowV2Classifier` and `KerasClassifier` classes just like is done with the PyTorch variation.

**To Reproduce**
Instantiate a `TensorFlowV2Classifer` or `KerasClassifier` instance using the `SparseCategoricalCrossentropy` loss. Call the `compute_loss(x, y)` function and use labels for `y` (e.g., shape `(n,)`) and there will be an error due the argmax function call.

**Expected behavior**
The labels should be validated and transformed accordingly.

**System information (please complete the following information):**
 - OS
 - Python version
 - ART version or commit number
 - TensorFlow / Keras / PyTorch / MXNet version

### PR DESCRIPTION
**Pull request by f4str**
# Description

Call the `check_and_transform_label_format` function when calling the `compute_loss` function in the `TensorFlowV2Classifier` and `KerasClassifier` classes. This fixes the argmax issue when passing integer labels.

Fixes #1959 

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — APPROVED** [review]
Hi @f4str Thank you very much for discovering and fixing this bug! The solution looks good to me.

**Comment by f4str** [comment]
Consolidated to PR #1956 since this change can be properly tested there.
